ArcGISRuntimeSDKQt_CppSamples/Routing/ServiceArea/ServiceArea.cpp

```diff
@@ -155,7 +155,25 @@ void ServiceArea::solveServiceArea()
   if (!barriers.isEmpty())
     m_parameters.setPolylineBarriers(barriers);
 
-  m_task->solveServiceArea(m_parameters);
+  m_task->solveServiceAreaAsync(m_parameters).then(this, [this](const ServiceAreaResult& serviceAreaResult)
+  {
+    setBusy(false);
+    if (serviceAreaResult.isEmpty())
+    {
+      m_message = "No Serice Areas calculated!";
+      emit messageChanged();
```

Comment:
I realize this code is unchanged and simply ported, but this looks like it should return early in this case. optional, but give it a quick look to see if that's the case. If it starts turning into a larger amount of work, leave it.

Reply:
Returned early. The m_graphicParent is created whenever it is been used so we can return early.
Tested and there is no regression.